### DISCUSSION TITLE
Request for Permission to Create Derivative of CP Editor as a VSCode Extension

### DISCUSSION THREAD

**Discussion by ryclif** [cpeditor/cpeditor#1173] [Q&A] [answered]
CP Editor been incredibly useful to me in my contests and I find it excellent at what it does. 
However, sometimes I wish I could do all of this right in VS Code with all of my other extensions and keybinds. There is an [existing project](https://github.com/agrawal-d/cph) which does this but 
> CPH is in a steady-state. There are no new features planned for the extension. Only bug-fixes will be worked on.

as it does not have nearly as many features as CP Editor, I would like to create my own extension.

I haven't decided on the name yet, it will probably be something like CP Assistant and it will be open source under GPL v3. Most of the features will not be exactly the same, but some like "Open Contest" will have to be a carbon-copy. 

I would be grateful to receive your permission to proceed with the development this project. If yes, please let me know where and how The CP Editor team would like to be credited. I am also open to any other specific conditions that you might have.

**Comment by ouuan** [comment, marked as answer]
You have permission to be inspired by CP Editor's features even without our consent. You can even use our code given that your code is open-sourced under GPL v3 and CP Editor is appropriately credited.

I think a mention of CP Editor somewhere like in the README is enough?
